rlclientlib/logger/async_batcher.h

```diff
@@ -96,15 +101,14 @@ namespace reinforcement_learning { namespace logger {
 
   template<typename TEvent, template<typename> class TSerializer>
   int async_batcher<TEvent, TSerializer>::append(TEvent&& evt, api_status* status) {
-
+    update_event_with_index(std::move(evt));
```

Comment:
Just thought a bit more - looks like there is still a problem here:
Following sequence:
1) original_count = 1, queue = [{index = 1}]
2) update_event from thread 1:       event with index=2
3) update_event from thread 2:       event with index=3
4) push to queue from thread 2:      queue = [{index=1}, {index=3}]
5) push to queue from thread 1:      queue = [{index=1}, {index=3], {index=2}]
We got events unordered by indices which is bad since it breaks original_count calculation